test/dbpoolTest.js

```diff
@@ -216,4 +217,46 @@ describe('DBPool Class Tests', () => {
         });
       });
   });
+  describe('new connection event', () => {
+    it('executes a callback when new connection event occurs', (done) => {
+      const pool = new DBPool({ url: '*LOCAL' });
+      // register the listener
+      pool.on('new connection', (connection) => {
+        expect(connection).to.be.instanceof(Connection);
```

Comment:
How do we know this callback is actually called. Does the testing framework require an `expect()` statement be executed or can we have a false positive test result?

Reply:
Mocha knows to wait for [done()](https://mochajs.org/#asynchronous-code) to be called when its passed to the `it('...', (done) =>)`.

Before calling done we have the `expect()` assertion that ensures `connection` passed to the callback is an instance of Connection class.

If the callback is not called in a timely manner the test will timeout.  Or fail if the assertion is incorrect.

Reply:
Ok, sounds like we wouldn't have any false results here. Based on how you described it working I have another possible concern. During `pool.createConnection()`, the pool is initialized and multiple connections are created. When each is created, the "new connection" event is is emitted for them. Wouldn't that mean we have multiple calls to the done callback? How does Mocha handle this?

Reply:
[pool.createConnection](https://github.com/IBM/nodejs-idb-pconnector/blob/fcb24766a1e39c9e46c36a03cdb27b0f319905a7/lib/dbPool.js#L44) will only create one connection in the pool. So here we are assured to only call the callback once.

[pool.attach](https://github.com/IBM/nodejs-idb-pconnector/blob/fcb24766a1e39c9e46c36a03cdb27b0f319905a7/lib/dbPool.js#L148) initializes the pool if its empty based on the `incrementSize` option (Default is 8 connections). Also will add additional connections when none are available.

That is why in the [second test case](https://github.com/IBM/nodejs-idb-pconnector/blob/fcb24766a1e39c9e46c36a03cdb27b0f319905a7/test/dbpoolTest.js#L252) I was keeping track of the times the callback was called then calling done when count was 3.

[Calling done multiple times](https://mochajs.org/#detects-multiple-calls-to-done) would lead to an error

Reply:
Oh, gotcha. I was thinking `createConnection` would initialize the pool. Then I think the second test is valid, but I still think that setting the library list and validating it for each connection is outside that test scope.

Reply:
OK will remove the second test

Reply:
As I said, I still thing the test is valid, it just doesn't need to set the library list and validate it. All it needs to validate is that when `pool.attach` is executed, that the callback is called the same number of times that `incrementSize` is defined to.